contexts/WalletManagerContext.tsx

```diff
@@ -660,10 +668,52 @@ export function WalletManagerProvider({ children }: { children: ReactNode }) {
 
     fetchBalance();
 
-    rpcClient.addEventListener("utxos-changed", fetchBalance);
+    function checkIncomingUtxos(event: {
+      added: UtxoEntryReference[];
+      removed: UtxoEntryReference[];
+    }) {
+      if (event.added.length === 0) {
+        return;
+      }
+
+      const txId = event.added[0].outpoint.transactionId;
```

Comment:
Are all added utxos are from the same transaction? I believe it is not.

Reply:
I believe UTXOs can come from different addresses by they will be in the same transaction.